pkg/controller/kogitoapp/kogitoapp_controller_test.go

```diff
@@ -272,3 +298,238 @@ func TestKogitoAppWithResource(t *testing.T) {
 	}
 
 }
+
+func TestReconcileKogitoApp_updateKogitoAppStatus(t *testing.T) {
+	kogitoapp := createFakeKogitoApp()
+	kogitoapp.Status = v1alpha1.KogitoAppStatus{
+		Builds: v1alpha1.Builds{
+			Complete: []string{
+				"test-app",
+				"test-app-build",
+			},
+		},
+		Route: "http://localhost",
+		Conditions: []v1alpha1.Condition{
+			{
+				Type: v1alpha1.DeployedConditionType,
+			},
+		},
+	}
+	images := createFakeImages(kogitoapp.Name)
+	s := setupSchema(kogitoapp)
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      kogitoapp.Name,
+			Namespace: kogitoapp.Namespace,
+		},
+	}
+	buildconfigRuntime := &buildv1.BuildConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kogitoapp.Name,
+			Namespace: kogitoapp.Namespace,
+			Labels: map[string]string{
+				kogitores.LabelKeyAppName:   kogitoapp.Name,
+				kogitores.LabelKeyBuildType: string(kogitores.BuildTypeRuntime),
+			},
+		},
+	}
+	buildconfigS2I := &buildv1.BuildConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kogitoapp.Name + "-build",
+			Namespace: kogitoapp.Namespace,
+			Labels: map[string]string{
+				kogitores.LabelKeyAppName:   kogitoapp.Name,
+				kogitores.LabelKeyBuildType: string(kogitores.BuildTypeS2I),
+			},
+		},
+	}
+	buildList := &buildv1.BuildList{
+		Items: []buildv1.Build{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      kogitoapp.Name,
+					Namespace: kogitoapp.Namespace,
+					Labels: map[string]string{
+						kogitores.LabelKeyAppName:   kogitoapp.Name,
+						kogitores.LabelKeyBuildType: string(kogitores.BuildTypeRuntime),
+					},
+				},
+				Status: buildv1.BuildStatus{
+					Phase: buildv1.BuildPhaseComplete,
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      kogitoapp.Name + "-build",
+					Namespace: kogitoapp.Namespace,
+					Labels: map[string]string{
+						kogitores.LabelKeyAppName:   kogitoapp.Name,
+						kogitores.LabelKeyBuildType: string(kogitores.BuildTypeS2I),
+					},
+				},
+				Status: buildv1.BuildStatus{
+					Phase: buildv1.BuildPhaseComplete,
+				},
+			},
+		},
+	}
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kogitoapp.Name,
+			Namespace: kogitoapp.Namespace,
+		},
+		Spec: routev1.RouteSpec{
+			Host: "localhost",
+		},
+	}
+
+	clientfake := &kogitoclient.Client{
```

Comment:
Please use the CreateFakeClient method from the `pkg/test` instead.
EDIT: I refactored it to include Prometheus and Discovery in https://github.com/kiegroup/kogito-cloud-operator/pull/100. You can rebase after we merge it.

Reply:
corrected